js/sketchfab.js

```diff
@@ -119,7 +120,13 @@ SketchfabPlugin.prototype.onSubmit = function onSubmit() {
 	var modelId = url.substring(url.lastIndexOf('/') + 1);
 	var text = this.renderText(modelId);
 
-	this.insertAtCursor(text);
+	if ((this.mode || '').toLowerCase() === 'wysiwyg') {
```

Comment:
That sounds good! Using our events is a great way to integrate.

It's not that the other way is "bad" per se, it just doesn't account for a bunch of edge cases. It'll be the difference between accommodating 98% of Vanilla sites and 100%.